api/metric/api.go

```diff
@@ -16,66 +16,97 @@ package metric
 
 import (
 	"context"
+	"time"
 
 	"go.opentelemetry.io/api/core"
 	"go.opentelemetry.io/api/registry"
 	"go.opentelemetry.io/api/unit"
 )
 
-type MetricType int
+// Kind categorizes different kinds of metric.
+type Kind int
 
+//go:generate stringer -type=Kind
 const (
-	Invalid    MetricType = iota
-	Gauge                 // Supports Set()
-	Cumulative            // Supports Inc()
+	Invalid        Kind = iota
+	CumulativeKind      // Supports Add()
+	GaugeKind           // Supports Set()
+	MeasureKind         // Supports Record()
 )
 
-type Meter interface {
-	// TODO more Metric types
-	GetFloat64Gauge(ctx context.Context, gauge *Float64GaugeHandle, labels ...core.KeyValue) Float64Gauge
+// Recorder is the implementation-level interface to Set/Add/Record individual metrics.
+type Recorder interface {
+	// RecordAt allows the SDK to observe a single metric event
+	RecordAt(ctx context.Context, t time.Time, value float64)
```

Comment:
Are you saying people will have to do
```
c.Inc(ctx, 1)
```

rather than

```
c.Inc(1)
```

Reply:
Yes. The rationale is that this lets us associated the distributed context with the update, allowing us to record the update in a span or as a structured metric event with all of its labels, not only those defined in the handle.

Reply:
That totally makes sense to me, but it's very strange for a function to take a context and not **use** the context to time out or unblock itself.

It's like "Take this thing that implies timeouts and cancelation, but don't actually use it to timeout or cancel anything.  Just use it as a grabbag"

I wonder if it is better to take one of these.

```
type Valuer interface {
Value(key interface{}) interface{}
}
var _ context.Context = Valuer(nil)
```